### DISCUSSION TITLE
How to reset pomodoro counter

### DISCUSSION THREAD

**Discussion by Yphrea** [focustimerhq/FocusTimer#644] [Q&A]
Hello, 

I really like this pomodoro app, it does what I want and is easy to use and minimalistic. However, I am struggling with resetting it at the start of the day. I would like to reset the counter for number of pomodoros finished before a longer break at the start of each day, but I can't find how to do this. I have tried restarting the application but this doesn't help, it remembers where in the cycle it was when I quit. How can I reset the pomodoro counter?

Best regards,
Pernilla

**Comment by kamilprusko** [comment]
Hi!

Currently it can be forced by:
```
dbus-send --session --type=method_call --print-reply=literal --dest=org.gnome.Pomodoro /org/gnome/Pomodoro org.gnome.Pomodoro.Reset
```

It should reset after 1h of being idle, but it's not reliable...

I think it could be fixed, but ideally few internals needs to be redesigned - primarily the way it detects system suspend. I'll address this while porting the app to GTK 4 (its in progress).

**Comment by kamilprusko** [comment]
In master branch I've added a commandline option `gnome-pomodoro --reset`

**Reply by cirelli94** [reply]
@kamilprusko could be possible to set a number of pomodoro done instead? I often do one/two and then get the long break...

Maybe I could contribute adding a `gnome-pomodoro --set <n>` or `gnome-pomodoro --reset <n>`?

**Reply by kamilprusko** [reply]
In the upcoming wip/gtk4 branch it wouldn't be possible to alter history that simply. It would require specifying start and end times of the completed pomodoros...

Few alternatives:
1. You can force to take a short break instead of a long one
2. Running a custom session https://github.com/gnome-pomodoro/gnome-pomodoro/issues/643
3. Altering history using UI
4. Stricter algorithm for taking breaks - currently it favours break/pomodoro ratio over specified number of cycles
5. In the upcoming version the manual reset will be more pronounced... So you could manually reset and take a long break when you need one
<img width="522" height="722" alt="Screenshot From 2025-09-30 10-07-49" src="https://github.com/user-attachments/assets/6b97f9c4-25f2-4ac7-bbc2-123efa8f7495" />

@cirelli94  Any of these alternatives would work for you?

**Reply by cirelli94** [reply]
I see from the screenshot that the pomodoro done will be visible, am I right?
<img width="344" height="428" alt="image" src="https://github.com/user-attachments/assets/58b16a69-aa25-485a-bb06-37fb84306f69" />
This is great and would be enough!
Thanks for your great work!

**Reply by kamilprusko** [reply]
Correct. It shows completed pomodoros
